src/node.cc

```diff
@@ -1559,12 +1575,14 @@ static void Abort(const FunctionCallbackInfo<Value>& args) {
 
 static void Chdir(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
+  CHECK(env->is_main_instance());
 
   if (args.Length() != 1 || !args[0]->IsString()) {
     return env->ThrowTypeError("Bad argument.");
   }
 
   node::Utf8Value path(args.GetIsolate(), args[0]);
+  ScopedLock::Write lock(&process_rwlock);
```

Comment:
Just wondering, is it necessary to take out the r/w lock when only the main isolate can call process.chdir()?